resetting_timer.go

```diff
@@ -62,6 +66,16 @@ func (NilResettingTimer) Time(func()) {}
 // Update is a no-op.
 func (NilResettingTimer) Update(time.Duration) {}
 
+// Percentiles panics.
+func (NilResettingTimer) Percentiles([]float64) []int64 {
+	panic("Percentiles called on a NilResettingTimer")
```

Comment:
Do we really want to call `panic` in these cases? I mean of course it's an obvious issue, but I wonder if a more "grateful" handling might be appropriate?

Reply:
This is how the approach that the go-metrics library is taking. The methods that you are not supposed to call for given types just panic.

You are only supposed to call `Percentiles` on a snapshot. I am also not mad about the lost type safety, but these types are behind the same interface, and I don't want to introduce more changes than necessary.